demos/icons/index.php

```diff
@@ -39,138 +38,156 @@
-			<a href="index.php" class="ui-button ui-button-inline ui-shadow ui-corner-all">Right <span class="ui-icon ui-icon-arrow-r"></span></a>
-			<a href="index.php" class="ui-button ui-button-inline ui-shadow ui-corner-all">Top<span class="ui-icon ui-icon-arrow-u ui-icon-display-block"></span></a>
-			<a href="index.php" class="ui-button ui-button-inline ui-shadow ui-corner-all">Bottom<span class="ui-icon ui-icon-arrow-d ui-icon-display-block"></span></a>
+			<a href="index.php" data-role="button" data-inline="true" data-icon="ui-icon-arrow-l">Beginning</a>
+			<a href="index.php" data-role="button" data-inline="true" data-icon="ui-icon-arrow-r" data-icon-position="end">End</a>
+			<a href="index.php" data-role="button" data-inline="true" data-icon="ui-icon-arrow-u" data-icon-position="top">Top</a>
+			<a href="index.php" data-role="button" data-inline="true" data-icon="ui-icon-arrow-d" data-icon-position="bottom">Bottom</a>
 		</div><!--/demo-html -->
 
-		<p>Inline mini buttons</p>
+		<h3>Inline mini buttons</h3>
 		
 		<div data-demo-html="true">
-			<a href="index.php" class="ui-button ui-button-inline ui-mini ui-shadow ui-corner-all"><span class="ui-icon ui-icon-arrow-l"></span> Left</a>
-			<a href="index.php" class="ui-button ui-button-inline ui-mini ui-shadow ui-corner-all">Right <span class="ui-icon ui-icon-arrow-r"></span></a>
-			<a href="index.php" class="ui-button ui-button-inline ui-mini ui-shadow ui-corner-all">Top<span class="ui-icon ui-icon-arrow-u ui-icon-display-block"></span></a>
-			<a href="index.php" class="ui-button ui-button-inline ui-mini ui-shadow ui-corner-all">Bottom<span class="ui-icon ui-icon-arrow-d ui-icon-display-block"></span></a>
+			<a href="index.php" data-role="button" data-inline="true" data-mini="true" data-icon="ui-icon-arrow-l">Beginning</a>
+			<a href="index.php" data-role="button" data-inline="true" data-mini="true" data-icon="ui-icon-arrow-r" data-icon-position="end">End</a>
+			<a href="index.php" data-role="button" data-inline="true" data-mini="true" data-icon="ui-icon-arrow-u" data-icon-position="top">Top</a>
+			<a href="index.php" data-role="button" data-inline="true" data-mini="true" data-icon="ui-icon-arrow-d" data-icon-position="bottom">Bottom</a>
 		</div><!--/demo-html -->
 
-		<h2>Icon-only</h2>
-
-		<p>Use "notext" as value for icon position if you want to create an icon-only button.</p>
-
-        <div data-demo-html="true">
-            <a href="index.php" class="ui-button ui-shadow ui-corner-all ui-button-icon-only">Delete<span class="ui-icon ui-icon-delete"></span></a>
+		<h2>Float icons</h2>
+		
+		<p>The framework offers helper classes to make icons inside block buttons float to the beginning or end.</p>
+		
+		<div data-demo-html="true">
+			<a href="index.php" data-role="button" data-icon="ui-icon-caret-l" data-classes="{ 'ui-button-icon': 'ui-button-icon ui-icon ui-icon-caret-l ui-widget-icon-floatbeginning' }">Beginning</a>
```

Comment:
Adding the helper class to the icon span with the classes option doesn't work (yet).
We should make that work or otherwise remove these examples.

See also https://github.com/jquery/jquery-mobile/issues/8145#issuecomment-127011501